ejb3/src/main/java/org/jboss/as/ejb3/remote/AssociationImpl.java

```diff
@@ -246,6 +240,33 @@ public SecurityIdentity getSecurityIdentity() {
         return cancellationFlag::cancel;
     }
 
+    private void updateAffinities(InvocationRequest invocationRequest, Map<String, Object> attachments, EJBLocator<?> ejbLocator, ComponentView componentView) {
+        Affinity legacyAffinity = null;
+        Affinity weakAffinity = null;
+        Affinity clusterAffinity = getClusterAffinity();
+
+        if (ejbLocator.isStateful() && componentView.getComponent() instanceof StatefulSessionComponent) {
+            final StatefulSessionComponent statefulSessionComponent = (StatefulSessionComponent) componentView.getComponent();
+            weakAffinity = legacyAffinity = getWeakAffinity(statefulSessionComponent, ejbLocator.asStateful());
+        } else if (componentView.getComponent() instanceof StatelessSessionComponent) {
+            // V3 and less used cluster affinity as a weak affinity for SLSBs
+            legacyAffinity = clusterAffinity;
+        }
+
+        // Always use the cluster as the strong affinity, if there is one
+        if (clusterAffinity != null) {
```

Comment:
I don't understand why "cluster affinity" needs special treatment.  SLSBs don't need a strong affinity to the cluster, just a weak one, as there is no state that requires it to use a clustered node.